### PR TITLE
Field of view coordinates and IRF axis names

### PR DESCRIPTION
**Pull request by cdeil**
The spec for field of view coordinates still needs to be written:
http://gamma-astro-data-formats.readthedocs.org/en/latest/info/coordinates.html#field-of-view

Also, I've started to write up recommended names for IRF coordinate axes:
http://gamma-astro-data-formats.readthedocs.org/en/latest/irfs/index.html#irf-axes
This info should probably be moved onto the coordinates page.

The hardest part is to describe / fix the FOV coordinates and names.
I'll make a proposal in the next few days and then we can discuss ...

### CODE REVIEW

**Comment by cdeil** [comment]
@lmohrmann - This issue is the last substantial topic to finalise the v0.2 of the spec, that we need for the HESS release. I'd like to propose again to drop ``DETX`` and ``DETY`` from the HESS events table. IMO it's just a waste of space, and given that we don't have agreement on it's definition (we put ALTAZ aligned, ctools assumes RADEC aligned) it just creates confusion / problems. Both Gammapy and ctools work perfectly fine with EVENTS that don't have a DETX / DETY column.

@TarekHC - Please check what you fill in ``DETX`` / ``DETY`` for MAGIC and also consider removing it, or document the orientation clearly.

@mackaiver - I see that for FACT you're already not adding these two columns to your DL3 EVENTS files. Well done!

Please review the attached commit, and directly edit via a follow-up commit if something is incorrect or unclear.

For now, I just tried to clearly describe the different FOV coordinates in use, but leave the decision what to use open, stating that this will be made by CTA hopefullly soon (cc @kosack from CTA / ctapipe).

@jknodlseder - Maybe you want to comment on this one as well?

File: source/events/events.rst [hunk 1]
```diff
@@ -45,31 +45,24 @@ Optional columns
...
 * ``DETX`` tform: ``1E``, unit: deg
-    * Reconstructed event X-coordinate in detector system
-      (nominal system, see :ref:`coords-fov`).
-      See also the `OGIP event list`_ standard.
+    * Reconstructed field of view X (see :ref:`coords-fov`).
 * ``DETY`` tform: ``1E``, unit: deg
-    * Reconstructed event Y-coordinate in detector system
-      (nominal system, see :ref:`coords-fov`).
-      See also the `OGIP event list`_ standard.
+    * Reconstructed field of view Y (see :ref:`coords-fov`).
```
Comment (lmohrmann) [inline source/events/events.rst]: Shouldn't DETX and DETY be replaced by the more clearly named FOV_ALTAZ_LON, etc...?

File: source/irfs/full_enclosure/bkg/index.rst [hunk 2]
```diff
@@ -58,7 +58,7 @@ Required columns:
 * ``ENERG_LO``, ``ENERG_HI`` -- ndim: 1, unit: TeV
     * Reconstructed energy axis
 * ``DETX_LO``, ``DETX_HI``, ``DETY_LO``, ``DETY_HI`` -- ndim: 1, unit: deg
-    * Field of view coordinates binning, see :ref:`coords-fov`
+    * Field of view coordinates binning (see :ref:`coords-fov`)
```
Comment (lmohrmann) [inline source/irfs/full_enclosure/bkg/index.rst]: These are of course also somewhat ambiguous, it's not clear along which system these are aligned. Add a note that this should be specified clearly?

**Comment by lmohrmann** [comment]
@cdeil Once we have a decision here, I can remove `DETX` and `DETY` from the event columns in the HAP exporter.

**Comment by cdeil** [comment]
> I'm OK with dropping DETX and DETY from the event column in general. Are you 100% sure it's not being used by ctools at the moment though?

They might be accessed and used when using a bkg3d model. I don't know. Can you or someone in Erlangen try? (I don't have ctools installed and don't want to spend time on it).

**Comment by lmohrmann** [comment]
> They might be accessed and used when using a bkg3d model. I don't know. Can you or someone in Erlangen try?

We'll have a look

File: source/general/coordinates.rst [hunk 3]
```diff
@@ -131,47 +131,76 @@ Astropy, i.e. this is without refraction corrections):
...
+different from the IACT camera coordinate detectors), it wasn't clear what to put,
+both ``(DETX, DETY) = (FOV_ALTAZ_LON, FOV_ALTAZ_LAT)``
+and ``(DETX, DETY) = (FOV_RADEC_LON, FOV_RADEC_LAT)``
+and very early on even TAN projections were used.
+
+Given this situation that there is no concensus yet, one suggestion is to avoid
+putting FOV coordinates in EVENTS, or if they are added, to clearly state how
+they are defined. This still leaves the problem with the background models,
+in case they are non-radially symmetric. We expect CTA to make a decision and to define
+FOV coordinate systems soon, to resolve this issue.
```
[the remark this answers is not in the document]
Comment (cdeil) [reply source/general/coordinates.rst]: I'm -1 on encouraging the use of new column names such as FOV_ALTAZ_LON, FOV_ALTAZ_LAT and starting to add support for those in exporters and Gammapy / ctools.
We weren't able to agree with DETX / DETY even while no-one was using it, so it's highly unlikely that we're going to agree on new column names.
I really suggest to leave this decision to the future and to CTA, we can't resolve this here while the relevant people aren't participating in the discussion and no-one has decision making power.

File: source/events/events.rst (hunk 1 above)
Comment (cdeil) [reply source/events/events.rst]: See my comment above.

File: source/irfs/full_enclosure/bkg/index.rst (hunk 2 above)
Comment (cdeil) [reply source/irfs/full_enclosure/bkg/index.rst]: I have written long comments / notes in the general FOV section. Does it really help to partly re-hash those comments here?
If you think so, would you be willing to add a commit to this branch with any text changes you like?

**Comment by cdeil** [comment]
Just to comment: on the Gammapy side, the plan is to only access RA / DEC / TIME and to compute FOV coordinates from that, and then to interpret background model DETX / DETY with ALTAZ alignment, i.e. how we fill it at the moment in HESS.
And then once CTA makes a decision (e.g. in 2019), change the supported format and coordinate definitions to that.

**Comment by lmohrmann** [comment]
@cdeil It appears ctools does use `DETX` and `DETY`, albeit only to compute offset angles that are used to query the IRFs event-wise. Maybe someone involved in ctools development could verify this? As I said, I'd be OK with removing these columns if none of the tools uses them. I'd still vote for including both the Ra/Dec and Alt/Az aligned FOV coordinates in the HESS test release, but that's not at issue here, right?

File: source/general/coordinates.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (lmohrmann) [reply source/general/coordinates.rst]: OK, fine with me!

File: source/irfs/full_enclosure/bkg/index.rst (hunk 2 above)
Comment (lmohrmann) [reply source/irfs/full_enclosure/bkg/index.rst]: Agreed, this is probably not necessary. Ignore my comment.

**Comment by cdeil** [comment]
@dtiziani confirmed that ctools runs fine without `DETX` / `DETY` present, so I'll remove it from the HESS release files EVENTS tables.

Merging this PR.

**Pull request merged**